### PR TITLE
Add new case: boot guest with the same mac address

### PR DESCRIPTION
**Pull request by kimi1978wy**
Boot guest with 2 virtio-net devices with the same mac address.

id: 1927120

Signed-off-by: Yu Wang <wyu@redhat.com>

### CODE REVIEW

**Comment by kimi1978wy** [comment]
@PaulYuuu and @yanglei-rh  Could you help to review this patch? Thanks a lot

 (1/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.same_mac_address.q35: PASS (169.59 s)
 (2/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.same_mac_address.q35: PASS (129.52 s)
 (3/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2016.x86_64.io-github-autotest-qemu.same_mac_address.q35: PASS (96.38 s)
 (4/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win2012.x86_64.io-github-autotest-qemu.same_mac_address.q35: PASS (171.67 s)
 (5/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.i386.1.io-github-autotest-qemu.same_mac_address.q35: PASS (288.83 s)
 (6/6) Host_RHEL.m9.u0.qcow2.virtio_scsi.up.virtio_net.Guest.Win8.x86_64.1.io-github-autotest-qemu.same_mac_address.q35: PASS (135.77 s)

File: qemu/tests/same_mac_address.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    1) Boot guest with the same mac address
+    2) Check if the driver is installed and verified
+    3) Check ip of guest
+    4) Ping out
+
+    :param test: kvm test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    timeout = int(params.get("timeout", 360))
```
Comment (PaulYuuu) [inline qemu/tests/same_mac_address.py:24]: `params.get_numeric`

File: qemu/tests/same_mac_address.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    vm = env.get_vm(params["main_vm"])
+    session = vm.wait_for_login(timeout=timeout)
+    error_context.context("Check if the driver is installed and "
+                          "verified", logging.info)
+    driver_name = params.get("driver_name", "netkvm")
+    session = utils_test.qemu.windrv_check_running_verifier(session, vm,
+                                                            test,
+                                                            driver_name,
+                                                            timeout)
+    time.sleep(60)
```
Comment (PaulYuuu) [inline qemu/tests/same_mac_address.py:36]: Could you explain the reason for this?

File: qemu/tests/cfg/same_mac_address.cfg [hunk 3]
```diff
@@ -0,0 +1,8 @@
+- same_mac_address:
+    only Windows
+    virt_test_type = qemu
+    type = same_mac_address
+    driver_name = "netkvm"
```
Comment (yanglei-rh) [inline qemu/tests/cfg/same_mac_address.cfg:5]: Can you add parameter  "only virtio_net" ?

**Comment by kimi1978wy** [comment]
(1/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.same_mac_address.q35: PASS (191.59 s)
 (2/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.same_mac_address.q35: PASS (187.36 s)
 (3/3) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.same_mac_address.q35: PASS (215.05 s)

**Comment by kimi1978wy** [comment]
@PaulYuuu and @yanglei-rh  pls help to review it again

**Comment by yanglei-rh** [comment]
LGTM

Test Pass:
(1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.x86_64.io-github-autotest-qemu.same_mac_address.q35: PASS (248.22 s)
 (1/1) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.same_mac_address.q35: PASS (260.83 s)
(2/2) Host_RHEL.m8.u4.product_av.qcow2.virtio_scsi.up.virtio_net.Guest.Win10.i386.io-github-autotest-qemu.same_mac_address.q35: PASS (306.53 s)

File: qemu/tests/same_mac_address.py (hunk 2 above)
Comment (PaulYuuu) [reply qemu/tests/same_mac_address.py:36]: Okay, please comment it, other LGTM.

**Review by PaulYuuu — APPROVED** [review]

**Pull request merged**